sympy/physics/mechanics/body.py

```diff
@@ -0,0 +1,167 @@
+
+        if mass is None:
+            _mass = Symbol(name + '_mass')
+        else:
+            _mass = mass
+
+        if frame is None:
+            self._frame = ReferenceFrame(name + '_frame')
+        else:
+            self._frame = frame
+
+        if body_inertia is None and mass is None:
+            _inertia = (inertia(self._frame, 1, 1, 1), self._masscenter)
+        else:
+            _inertia = (body_inertia, self._masscenter)
+
+        self._masscenter.set_vel(self._frame, 0)
+
+        # If user passes masscenter and mass then a particle is created
+        # otherwise a rigidbody. As a result a body may or may not have inertia.
+        if body_inertia is None and mass is not None:
+            Particle.__init__(self, name, self._masscenter, _mass)
+        else:
+            RigidBody.__init__(self, name, self._masscenter, self._frame, _mass, _inertia)
+
+    def add_force(self, force_vector, point=None, frame=None):
+        """
+        Adds force to the body by adding a force vector in form of a tuple
+        of length 3 and a point on which the force will be applied.
+
+        Parameters
+        ---------
+        force_vector: A 3-Tuple
+            Defines the force vector w.r.t frame passed (or Body's frame). The
+            tuple defines the values of vector in x, y and z directions of the
+            frame.
+        point: Point
+            Defines the point on which the force must be applied.
+        frame: ReferenceFrame
+            Defines the frame w.r.t which force vector must be defined.
+
+        Example
+        --------
+        To add a force of magnitude 1 in y direction on a point at distance of
+        1 in x direction. All the directions are w.r.t body's frame.
+
+        >>> from sympy import Symbol
+        >>> from sympy.physics.mechanics import Body
+        >>> body = Body('body')
+        >>> g = Symbol('g')
+        >>> body.add_force((body.get_mass() * g, 0, 0), body.get_masscenter())
+
+        To define the force_vector w.r.t some other frame, pass the frame as well.
+
+        >>> from sympy import Symbol
+        >>> from sympy.physics.mechanics import Body
+        >>> parent = Body('parent')
+        >>> child = Body('child')
+        >>> g = Symbol('g')
+        >>> frame = parent.get_frame()
+        >>> masscenter = child.get_masscenter()
+        >>> gravity = child.get_mass() * g
+        >>> body.add_force((gravity, 0, 0), masscenter, frame)
+
+        """
+        if point is None:
+            point = self._masscenter  # masscenter
+
+        if frame is None:
+            frame = self._frame
+
+        if not isinstance(force_vector, tuple):
+            raise TypeError("Force vector must be a tuple of length 3")
+        else:
+            force_vector = self._convert_tuple_to_vector(force_vector, frame)
+        self.force_list.append((point, force_vector))
+        self._counter += 1
+
+    def _convert_tuple_to_vector(self, pos_tuple, frame):
```

Comment:
In the most general sense, forces are applied to points in a system. You have a Body class and you want to define a concept of adding a force to a body. A body has a point, the mass center, so you only need a vector. This vector should be defined in any frame or combination of frames. What you have right now is very limiting and a bit contrived. You can also have the point arg which will define a new point to apply the force wrt to the mass center in the body's frame. So the API would be more like:

```python
def apply_force(vec, point=None):
```

I chose "apply" here instead of "add" because forces can actually be added, but I don't think that is what this method is actually doing.